src/unistd.rs

```diff
@@ -1078,8 +1078,8 @@ pub fn gethostname() -> Result<OsString> {
 /// let f = tempfile::tempfile().unwrap();
 /// close(f.into_raw_fd()).unwrap(); // Good.  into_raw_fd consumes f
 /// ```
-pub fn close(fd: RawFd) -> Result<()> {
-    let res = unsafe { libc::close(fd) };
+pub fn close<Fd: IntoRawFd>(fd: Fd) -> Result<()> {
```

Comment:
Is this change really supposed to be part of this PR, or a different one?

Reply:
You are right, this was not supposed to be in this PR. Removed it.